packages/wallet/src/SingleAddressWallet.ts

```diff
@@ -288,6 +295,24 @@ export class SingleAddressWallet implements ObservableWallet {
       tip$: this.tip$,
       transactionsHistoryStore: stores.transactions
     });
+
+    this.#rollbackSubscription.add(
```

Comment:
@mkazlauskas , do you think a test is needed for this code too?

Reply:
Yes, that would be great - I suggest `test/SingleAddressWallet/rolback.test.ts`

You can inject stub ChainHistoryProvider with multiple mockResolvedValueOnce and mock a TxSubmitProvider. Then await for `firstValueFrom(transactions.history$.pipe(filter(txs => txs.length === n)))` before asserting calls on the TxSubmitProvider

Reply:
Done here: [afa2d65](https://github.com/input-output-hk/cardano-js-sdk/pull/351/commits/afa2d650c5e559cf6ad8847d6fc89ace1f4827e1)